File: 02_algorithm/programmers/Level2/타겟 넘버.py
def solution(numbers, target):
    answer_list = [0]
    for i in numbers:
        temp_list = []
        for j in answer_list:
            temp_list.append(j + i)
            temp_list.append(j - i)
        answer_list = temp_list
    answer = answer_list.count(target)
    return answer

print([1, 1, 1, 1, 1], 3)

File: 06_js/04_algorithm/programmers/Level2/타겟 넘버.js
function solution(numbers, target) {
  let answer = 0
  let answerArr = [0]
  for (let i = 0; i < numbers.length; i++) {
    let tempArr = new Array()
    for (let j = 0; j < answerArr.length; j++) {
      tempArr.push(answerArr[j] + numbers[i], answerArr[j] - numbers[i])
    }
    answerArr = tempArr
  }
  return answerArr.filter(num => target === num).length
}

console.log(solution([1, 1, 1, 1, 1], 3))